receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -134,16 +134,15 @@ func (tr *transaction) Commit() error {
 		// never added any data points, that the transaction has not been initialized.
 		return nil
 	}
-
 	metrics, numTimeseries, droppedTimeseries, err := tr.metricBuilder.Build()
 	observability.RecordMetricsForMetricsReceiver(tr.ctx, numTimeseries, droppedTimeseries)
 	if err != nil {
 		return err
 	}
+	if tr.jobsMap != nil {
```

Comment:
Can you please add a comment to explain the behavior here? It is not clear what and how this PR changes, why it was a problem before and why it is correct now.

Reply:
I added a comment in the code. If 'metrics' is empty, we shouldn't pass it on to ConsumeMetricsData() but the call to the metrics adjuster could potentially return an empty slice. Moving the call before the empty check resolves the issue.

Reply:
I think this does not belong to this PR. Can we have a different PR that changes this logic.

Reply:
Removed - I'll send another pr shortly.